### DISCUSSION TITLE
How to write "exec 3<>/path/to/named/pipe"

### DISCUSSION THREAD

**Discussion by rharish101** [fish-shell/fish-shell#9672] [Q&A] [answered]
I'm trying to port one of my zsh scripts to fish, and I came across this issue. How do I write the following command in fish:
```sh
exec 3<>/path/to/named/pipe
```

I want to write to a named pipe without waiting for readers, as well as not stop reading when there aren't any writers available.

Here's the reference in the [man page for fifo](https://man.archlinux.org/man/fifo.7):
```
Under Linux, opening a FIFO for read and write will succeed both in blocking and nonblocking mode. POSIX leaves this behavior undefined. This can be used to open a FIFO for writing while there are no readers available. A process that uses both ends of the connection in order to communicate with itself should be very careful to avoid deadlocks.
```

**Comment by faho** [comment, marked as answer]
Fish does not have global redirections like that, so you don't. This is #3948.
